From 1b9bc1583d882866d64420df39bf5df4bce22bd6 Mon Sep 17 00:00:00 2001
Message-Id: <1b9bc1583d882866d64420df39bf5df4bce22bd6.1307392642.git.dvhart@linux.intel.com>
In-Reply-To: <e5e52482147151aaaafbd388d1e5978268e51d24.1307392642.git.dvhart@linux.intel.com>
References: <e5e52482147151aaaafbd388d1e5978268e51d24.1307392642.git.dvhart@linux.intel.com>
From: Darren Hart <dvhart@linux.intel.com>
Date: Mon, 6 Jun 2011 10:19:00 -0700
Subject: [PATCH 2/2] board-omap3beagle: allow for building without wl1271

While the header file wl12xx.h is ifdef'd to include if the wl1271 driver is
built, the init routine calls into it regardless. Ideally, the module would
perform its own initialization at load time and we wouldn't need to ifdef these
calls in the general board initialization. For now, follow the existing practice
in this file and ifdef the wl1271 init block.

Signed-off-by: Darren Hart <dvhart@linux.intel.com>
---
 arch/arm/mach-omap2/board-omap3beagle.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/board-omap3beagle.c b/arch/arm/mach-omap2/board-omap3beagle.c
index b618cb6..caaed82 100644
--- a/arch/arm/mach-omap2/board-omap3beagle.c
+++ b/arch/arm/mach-omap2/board-omap3beagle.c
@@ -937,6 +937,7 @@ static void __init omap3_beagle_init(void)
 		gpio_export(162, 1);
 	}
 
+#if defined(CONFIG_WL1271) || defined(CONFIG_WL1271_MODULE)
 	if(!strcmp(expansionboard_name, "bbtoys-wifi"))
 	{
 		if (wl12xx_set_platform_data(&omap_beagle_wlan_data))
@@ -944,6 +945,7 @@ static void __init omap3_beagle_init(void)
 		printk(KERN_INFO "Beagle expansionboard: registering wl12xx platform device\n");
 		platform_device_register(&omap_vwlan_device);
 	}
+#endif
 
 	usb_musb_init(&musb_board_data);
 	usb_ehci_init(&ehci_pdata);
-- 
1.7.1

